### PR TITLE
T team statistics iteration

### PR DESCRIPTION
**Pull request by tony-gomes**
All working methods for Iteration 5

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
       CSV.foreach(file, csv_options) { |row| item_class.add(item_class.new(row.to_hash)) }
   end
 
+  def total_number_games_by_team_id?(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:27]: Helper query method for total number of games by team_id

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+    games = GameTeam.all
+    total_games = 0
+
+    games.each do |game|
+      total_games += 1 if game.last.key?(team_id)
+    end
+    total_games
+  end
+
+  def all_game_teams_by_team_id(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:37]: Helper utility method returns all GameTeam Objects that match team_id

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+    game_teams = GameTeam.all
+    all_game_teams = []
+
+    game_teams.each do |game|
+      all_game_teams << game.last[team_id] if game.last.key?(team_id)
+    end
+    all_game_teams
+  end
+
+  def all_games_by_team_id(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:48]: Helper utility method returns all Game Objects that match team_id

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+    return -0.0 if total_games.zero?
+
+    total_wins = total_results_by_team_id(team_id).map do |game|
+      game == "WIN" ? 1 : nil
+    end.compact.sum
+
+    (total_wins / total_games).round(2)
+  end
+
+  def all_goals_scored_by_team_id(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:78]: All goals scored by team_id

File: lib/stat_tracker.rb [hunk 5]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+    all_game_teams_by_team_id(team_id).map do |game|
+      game.goals
+    end
+  end
+
+  def score_differences_by_team_id(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:85]: All differences between team_id & opponent across Game

File: lib/stat_tracker.rb [hunk 6]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+      if game.last.home_team_id == team_id
+        goal_differences << (game.last.home_goals - game.last.away_goals)
+      elsif game.last.away_team_id == team_id
+        goal_differences << (game.last.away_goals - game.last.home_goals)
+      end
+    end
+    goal_differences
+  end
+
+  def get_team_name(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:100]: Helper method for getting team name by team_id

File: lib/stat_tracker.rb [hunk 7]
```diff
@@ -18,4 +18,287 @@ def self.create_items(file, item_class)
...
+
+  def biggest_team_blowout(team_id)
+    score_differences_by_team_id(team_id).max
+  end
+
+  def worst_loss(team_id)
+    score_differences_by_team_id(team_id).min.abs
+  end
+
+  def all_team_average_wins_by_opponent(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb]: Helper method for averages wins across all games by team_id

File: lib/stat_tracker.rb [hunk 8]
```diff
@@ -18,4 +18,157 @@ def self.create_items(file, item_class)
...
+    end
+    goal_differences
+  end
+
+  def get_team_name(team_id)
+    team_id = team_id.to_i if team_id.class != Integer
+    Team.all[team_id].team_name
+  end
+
+  def win_percentage_by_season(team_id)
```
Comment (tony-gomes) [inline lib/stat_tracker.rb:105]: All win percentages across seasons by team_id

File: lib/stat_tracker.rb (hunk 6 above)
Comment (DavidTTran) [reply lib/stat_tracker.rb:100]: I think I have a similar helper method called `return_team_name` not something that has to be changed but just as a note for refactoring

**Pull request merged**